src/ucp/proto/proto_select.c

```diff
@@ -556,6 +556,59 @@ ucp_proto_select_elem_init_thresh(ucp_worker_h worker,
     return status;
 }
 
+/**
+ * Get map of lanes used in the selected protocols.
+ */
+static ucp_lane_map_t
+ucp_proto_select_get_lane_map(ucp_worker_h worker,
+                              const ucp_proto_select_elem_t *select_elem)
+{
+    ucp_lane_map_t lane_map = 0;
+    size_t range_start, range_end;
+    ucp_proto_query_attr_t query_attr;
+    int proto_valid;
+
+    range_end = -1;
+    do {
+        range_start = range_end + 1;
+        proto_valid = ucp_proto_select_elem_query(worker, select_elem,
```

Comment:
maybe assert that lane_map is not 0 inside ucp_proto_select_elem_query?
it's never expected to be 0 and it can catch issue that some protocol is not setting it

Reply:
Added assert here [1321f71](https://github.com/openucx/ucx/pull/8986/commits/1321f716b6909cc8fa75427b1f42dc796896e561)